itests/src/test/java/fi/metatavu/edelphi/test/ui/base/index/ChangePlanTestsBase.java

```diff
@@ -0,0 +1,242 @@
+    planMocker.mock();
+  }
+  
+  @After
+  public void afterTest() {
+   deleteOrderHistories();
+    planMocker.cleanup();
+  }
+  
+  @Test
+  public void testBasicToPlus() {
+    testPlanChange(null, BASIC_TYPE, BASIC_NAME, null, 1, 100, PLUS_TYPE, PLUS_NAME, 100, null, null);
+    testPlanChange(null, BASIC_TYPE, BASIC_NAME, null, 2, 200, PLUS_TYPE, PLUS_NAME, 180, null, null);
+  }
+
+  @Test
+  public void testBasicToPremium() {
+    testPlanChange(null, BASIC_TYPE, BASIC_NAME, null, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, null, null);
+    testPlanChange(null, BASIC_TYPE, BASIC_NAME, null, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, null, null);
+  }
+  
+  @Test
+  public void testPlusToPlus() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 1, 100, PLUS_TYPE, PLUS_NAME, 100, null, 30);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 2, 200, PLUS_TYPE, PLUS_NAME, 180, null, 30);
+  }
+  
+  @Test
+  public void testPlusToPremiumNoCompensation() {
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, null, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, null, null);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, null, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, null, null);
+  }
+  
+  @Test
+  public void testPlusToPremiumWithCompensation() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, 26.1d, null);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, 26.1d, null);
+  }
+  
+  @Test
+  public void testPremiumToPremium() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(3), PREMIUM_TYPE, PREMIUM_NAME, oldEnds, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, null, 30);
+    testPlanChange(planMocker.getPlanId(3), PREMIUM_TYPE, PREMIUM_NAME, oldEnds, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, null, 30);
+  }
+  
+  @Test
+  public void testCancelPayment() {
+    Date oldSubscriptionEnd = Date.from(OffsetDateTime.now().plusDays(30).toInstant());
+    Double newPlanPrice = 200d;
+    Double compensation = 26.1d;
+    DateFormat dateFormat = DateFormat.getDateInstance(DateFormat.MEDIUM);
+    String oldEndDateFormatted = oldSubscriptionEnd != null ? dateFormat.format(oldSubscriptionEnd) : null;
+    Double totalPrice = Math.max(newPlanPrice - compensation, 0);
+    int newPlanIndex = 3;
+    String oldPlanName = "Plus";
+    
+    NumberFormat currencyFormatter = NumberFormat.getCurrencyInstance();
+    currencyFormatter.setCurrency(Currency.getInstance("EUR"));
+    String compensationFormatted = compensation != null ? currencyFormatter.format(compensation) : null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2583)